File: Packages/module-ai/Sources/ModuleAI/Application/UseCases/AIChat/LoadTemplateUseCase.swift
//
//  LoadTemplateUseCase.swift
//  ModuleAI
//
//  Created by Codex on 2026/1/8.
//

import Foundation

struct LoadTemplateUseCase {
    private let repository: AIChatRepository

    init(repository: AIChatRepository) {
        self.repository = repository
    }

    func execute() async throws -> LoadTemplateOutput {
        let template = try await repository.loadTemplate()
        return LoadTemplateOutput(questions: template.questions)
    }
}

File: app-android/feature/ai-chat/src/main/java/com/aidatainsight/android/feature/aichat/application/usecase/LoadTemplateUseCase.kt
package com.aidatainsight.android.feature.aichat.application.usecase

import com.aidatainsight.android.feature.aichat.application.model.LoadTemplateOutput
import com.aidatainsight.android.feature.aichat.domain.AIChatRepository

class LoadTemplateUseCase(
    private val repository: AIChatRepository,
) {
    suspend operator fun invoke(): LoadTemplateOutput {
        val template = repository.loadTemplate()
        return LoadTemplateOutput(questions = template.questions)
    }
}
